library/alloc/src/collections/btree/map.rs

```diff
@@ -490,7 +490,9 @@ impl<K: fmt::Debug, V: fmt::Debug> fmt::Debug for RangeMut<'_, K, V> {
     }
 }
 
-impl<K, V> BTreeMap<K, V> {
+// N.B. bound not on function to avoid unnecessary
+// const bound on `Ord`: see #87375
+impl<K: Ord, V> BTreeMap<K, V> {
```

Comment:
this should not make a difference at all. All bounds on the impl must apply for functions defined inside, too.

if this truly fixes things, that is a bug. Instead we should use a `?const Ord` bound

Reply:
Hmmmm, are you saying bounds on impls should be treated as const bounds when inherent functions are const? That would break a lot of existing code I think.

```rust
impl<T: Clone> MyType {
    fn a() { ... }; // T: Clone
    const fn b() { ... }; // T: const Clone? 
}
```

Reply:
Only code using feature gates, stable code already doesn't allow this: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=428faddc9d2bce802ab7423fdfa7db33

Reply:
Wow, I really didn't know this. But can we use `?const` inside liballoc? That feature is marked as incomplete.

Reply:
hmm... I guess we are just going to break ppl using btreemap in const contexts? Or try to remove the bound in a separate PR first?

Reply:
From the crater run I think not many people are using this feature. I don't think we should remove the bound because it might require more work when re-adding the bound back. I'm fine with breaking BTree{Map,Set}::new for now.